packages/appframe/src/react/index.tsx

```diff
@@ -65,19 +70,62 @@ const styles = {
   topnav: () => css(stylesheet['.psds-appframe__topnav'])
 }
 
-const AppFrameContext = createContext()
+interface AppFrameContextValue {
+  closeSidenav: () => void
+  openSidenav: () => void
+  sidenavOpen: boolean | undefined
+  sidenavVariant: ValueOf<typeof vars.sidenavVariants>
+}
+const AppFrameContext = createContext<AppFrameContextValue>({
+  closeSidenav: () => {},
+  openSidenav: () => {},
+  sidenavOpen: undefined,
+  sidenavVariant: vars.sidenavVariants.closed
+})
+
+type RenderProp<P extends Record<string, unknown>> = (
+  props: P
+) => React.ReactNode
+
+interface AppFrameProps extends HTMLAttributes<HTMLDivElement> {
+  onRequestSideNavClose?: () => void
+  onRequestSideNavOpen?: () => void
+  sidenav?: ReactChild | RenderProp<{ visible: boolean }>
```

Comment:
i should probably use `ReactNode` instead of `ReactChild` throughout this file. will update